src/main/java/com/yelp/nrtsearch/server/luceneserver/field/ObjectFieldDef.java

```diff
@@ -0,0 +1,130 @@
+/*
+ * Copyright 2020 Yelp Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.yelp.nrtsearch.server.luceneserver.field;
+
+import com.google.gson.Gson;
+import com.yelp.nrtsearch.server.grpc.Field;
+import com.yelp.nrtsearch.server.luceneserver.AddDocumentHandler;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import org.apache.lucene.document.Document;
+
+public class ObjectFieldDef extends IndexableFieldDef {
+
+  private final Gson gson;
+  private final boolean isNested;
+
+  protected ObjectFieldDef(String name, Field requestField) {
+    super(name, requestField);
+    this.isNested = requestField.getNested();
+    gson = new Gson();
+  }
+
+  @Override
+  public String getType() {
+    return "OBJECT";
+  }
+
+  @Override
+  public void parseDocumentField(
+      Document document, List<String> fieldValues, List<List<String>> facetHierarchyPaths) {}
+
+  @Override
+  public void parseFieldWithChildren(
+      AddDocumentHandler.DocumentsContext documentsContext,
+      List<String> fieldValues,
+      List<List<String>> facetHierarchyPaths) {
+    if (!isNested) {
+      parseFieldWithChildren(documentsContext.getRootDocument(), fieldValues, facetHierarchyPaths);
+    } else {
+      List<Map<String, Object>> fieldValueMaps = new ArrayList<>();
+      fieldValues.stream()
+          .map(e -> gson.fromJson(e, Map.class))
+          .forEach(e -> fieldValueMaps.add(e));
+      List<Document> childDocuments =
+          fieldValueMaps.stream()
+              .map(e -> createChildDocument(e, facetHierarchyPaths))
+              .collect(Collectors.toList());
+      documentsContext.addChildDocuments(this.getName(), childDocuments);
+    }
+  }
+
+  /**
+   * create a new lucene document for each nested object
+   *
+   * @param fieldValue
+   * @param facetHierarchyPaths
+   * @return lucene document
+   */
+  private Document createChildDocument(
+      Map<String, Object> fieldValue, List<List<String>> facetHierarchyPaths) {
+    Document document = new Document();
```

Comment:
yea, I'll add them later when I am adding the query. It will be more clear whether these identifier are needed when there is a ToParentBlockJoinQuery test